compiler/test/dotty/tools/vulpix/ParallelTesting.scala

```diff
@@ -852,21 +852,21 @@ trait ParallelTesting extends RunnerOrchestration { self =>
     extends Test(testSources, times, threadLimit, suppressAllOutput) {
     protected def encapsulatedCompilation(testSource: TestSource) = new LoggedRunnable {
       def checkTestSource(): Unit = tryCompile(testSource) {
-        def fail(msg: String): Nothing = {
+        def fail(msg: String): Unit = {
           echo(msg)
           failTestSource(testSource)
-          ???
         }
         testSource match {
           case testSource@JointCompilationSource(_, files, flags, outDir, fromTasty, decompilation) =>
             val sourceFiles = testSource.sourceFiles
-            val reporter =
-              try compile(sourceFiles, flags, true, outDir)
-              catch {
-                case ex: Throwable => fail(s"Fatal compiler crash when compiling: ${testSource.title}")
-              }
-            if (reporter.compilerCrashed)
-              fail(s"Compiler crashed when compiling: ${testSource.title}")
+            try {
+              val reporter = compile(sourceFiles, flags, true, outDir)
+              if (reporter.compilerCrashed)
+                fail(s"Compiler crashed when compiling: ${testSource.title}")
+            }
+            catch {
```

Comment:
It should not happen, right? The exception is already caught in `def compile`: https://github.com/lampepfl/dotty/blob/c25fbea02bb3620fbec6f142fc6cbbbfdbeda5ed/compiler/test/dotty/tools/vulpix/ParallelTesting.scala#L374-L390

Reply:
In theory it should not happen. But if it does, we would report it and not crash.

Reply:
But if a crash happens, it is a bug in the test framework and a crash is arguably better than reporting a test failure.

Also, `compile` is called in multiple places in this file and that's the only place where it is wrapped in a try catch:
- https://github.com/lampepfl/dotty/blob/29aeb02be0c08a20373027a0d80deaa87cb699ff/compiler/test/dotty/tools/vulpix/ParallelTesting.scala#L589
- https://github.com/lampepfl/dotty/blob/29aeb02be0c08a20373027a0d80deaa87cb699ff/compiler/test/dotty/tools/vulpix/ParallelTesting.scala#L598
- https://github.com/lampepfl/dotty/blob/29aeb02be0c08a20373027a0d80deaa87cb699ff/compiler/test/dotty/tools/vulpix/ParallelTesting.scala#L685

Reply:
I agree with Allan

Reply:
Ok